appserver/web/war-util/pom.xml

```diff
@@ -120,4 +120,20 @@
             <artifactId>glassfish</artifactId>
         </dependency>
    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.github.siom79.japicmp</groupId>
+                <artifactId>japicmp-maven-plugin</artifactId>
+                <configuration>
+                    <parameter>
+                        <excludes>
+                            <exclude>org.glassfish.web.loader.CachingReflectionUtil</exclude>
```

Comment:
I believe this will break semantic versioning, this will require reworking your changes.

Reply:
This is an internal class. It was moved into a different module.  Nothing should be affected.

Reply:
We'd strongly prefer that you keep changes in line with semantic versioning - no breaking changes on minor version updates.
Parts of the codebase are shared with Enterprise and extensions, the semantic versioning is in place to help ensure compilation and runtime errors don't unknowingly propagate. 

We'll need to make sure this class is not used in any of the non-shared codebase (not just in this repo) before allowing this breaking change to go through.

Reply:
I suppose I could deprecate it. More changes to this API are coming from #7097 so stay tuned for that anyway

Reply:
I have made the changes to deprecate the class and remove the `japicmp` exclusion
Thank you